software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+﻿script "lazydogpile"
+/*
+This stack is mostly an empty "hull" that ties everything else together
+Some stuff, like the Livecode engine and objects, need to be binary
+Everything else is in text files
+*/
+local sMain -- main[doc][rec][tag]=data/pointers
+-- logic
+local sPointerTags -- space delimited list of pointer categories
+local sMainShort
+local sTopoSort
+local sNoteList
+local sNotVisited
+--  display
+local sNowDoc -- the document the user is currently focused on ! requires update
+local sNowItem -- cr delimited list of records currently selected ! requires update
+local sNowFocus -- ! from GUI test
+local sGraphDrawing -- ! from GUI test; sGraphData + layer info
+---- data[nodes][1...n][layer]=#
+---- data[abutFlow][1...n][=#,#,#,etc (nodes directly flowed to this one)
+---- data[layers][-n...0...n]=#,#,#,etc (nodes)
+---- data[visited][nodes/links]=#,#,#,etc
+local sGUIstuff -- ! from GUI test; graphics
+--- stuff[nodes][1...n][graphic]=livecode long id
+--- stuff[nodes][1...n][horz/vert]=# (pixels)
+--- stuff[nodes][1...n][type]=state/change/flow/etc
+--- stuff[nodes][1...n][abutFlow]=#,#,#,etc
+--- stuff[links][1...n][graphic]=livecode long id
+--- stuff[nodes/links][1...n][selected]=true
+local sNodeSize -- ! from GUI test; node graphic height and width
+local sLayerSize -- ! from GUI test; layer width
+local sLinkSize
+local s1Down -- empty/# (anchor where button first went down)
+local s1Enter -- empty/#/background (changes as mouse enters things, empty means never left anchor)
+local sMouseHistory -- state at the end of mouseTracker
+-- hist[1down]=true/false
+-- hist[inside]=empty/#/background
+-- mouseTracker and sMouseHistory cycle on every mouse message
+-- s1Down cycles on mouse up and down
+-- if s1Down is #, then s1Enter cycles on every area the mouse crosses over
+local sPreviewGraphics -- graphic preLinkLeft, field preNode, graphic PreLinkRight
+local sGraphData -- ! for GUI test; replace
+---- data[nodes][1...n][next/prev]=#,#,#,etc
+---- data[links][1...n][next/prev]=#
+---- data[lastRec]=#
+local sResources -- array[doc][input/output][rec]=weight
+local sScopeSketch -- array[doc][case][role]=rec,rec,rec... start at case 0, then each [top]
+--local sBusy -- added 15DEC2015 to block messages when a handler is still running; replaced with cBusy of stack howstrBeta 17FEB2016
+local sTooltipVisible -- ! this isn't actually being passed; the handler just checks the control itself
+local sNodeDetailsGeometry
+-- [startWidth] 400
+-- [startHeight] 400
+-- [tagDataList] 5,30,200,395
+-- [tagLabel] 205,30,395,51
+-- [editData] 205,56,395,395
+-- [loadImgButt] 219,41
+-- [clearImgButt] 256,41
+-- [expandImgButt] 300,41
+-- [setCaptureButt] 340,41
+-- [snapCaptureButt] 372,41
+local sCaptureAreaGeometry
+-- [captureAreaRect] 400,400,800,800 (width and height)
+-- square icons set to middle of height and equally spaced across width, 1/5th each
+
+on setupAsOf3FEB2016
```

Comment:
Any suggestions? I guess I figured if/when I made a different setup handler it would get a different name. Do you mean that dates should go into a comment instead of the handler name?